fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/temp/GraphExpansion.java

```diff
@@ -321,6 +328,16 @@ public SelectExpression buildSelect() {
             return new SelectExpression(RecordConstructorValue.ofColumns(resultColumns), quantifiers, getPredicates());
         }
 
+        @Nonnull
```

Comment:
Oh this method looks a bit subtle to me :)
So it requires that the object `overQuantifier` was added in some previous step to the `quantifiers` list, right?

Would it make sense to expose the list of quantifiers the builders has so far, and leave it to the consumer to decide what to do next?

e.g. something like:
```
public boolean hasQuantifier(@Nonnull final Quantifier.ForEach overQuantifier) {...}
```

Reply:
This is a short hand where you single out one quantifier to become the `*` in `select *`. I want to avoid pulling up all it's columns to just put them back together, i.e.
```
quantifier flows RECORD(a, b, c)
```
I don't want to create:
```
SELECT record_constructor(q.a as a, q.b as b, q.c as c)
FROM q
```

I would rather like to create:
```
SELECT quantified_object(q)
FROM q
```
which directly takes the entire record from `q` and makes it the output.

This is obviously an optimization for a specific case which we can possible also have transforms/rewrites for.